File: 3120. Count the Number of Special Characters I/Solution.cpp
#include <algorithm>
#include <array>
#include <ranges>
#include <string>
using namespace std;

class Solution {
public:
    int numberOfSpecialChars(string word)
    {
        array<unsigned, 26> seen { { 0 } };
        for (char c : word) {
            if (c >= 'a' && c <= 'z')
                seen[c - 'a'] |= 1;
            else
                seen[c - 'A'] |= 2;
        }
        return ranges::count(seen, 3);
    }
};

File: 3120. Count the Number of Special Characters I/Solution.py
class Solution:
    def numberOfSpecialChars(self, word: str) -> int:
        seen = [0] * 26
        for c in word:
            if "a" <= c <= "z":
                seen[ord(c) - ord("a")] |= 1
            else:
                seen[ord(c) - ord("A")] |= 2
        return seen.count(3)
